File: frontend/src/types/email.ts
export interface EmailLog {
  id: number;
  gmail_message_id: string;
  from_email: string | null;
  subject: string | null;
  source_label: string | null;
  doc_type: string | null;
  status: string | null;
  rule_applied_id: number | null;
  dropbox_folder_id: number | null;
  dropbox_file_path: string | null;
  error_message: string | null;
  processed_at: string;
}

export interface ProcessResult {
  total: number;
  procesado: number;
  sin_regla: number;
  error: number;
  skipped: number;
}

export interface EmailStats {
  total: number;
  procesado: number;
  sin_regla: number;
  error: number;
}

File: backend/app/schemas/email.py
"""Schemas Pydantic del historial de correos procesados (spec §8.7)."""
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class EmailLogOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    gmail_message_id: str
    from_email: str | None = None
    subject: str | None = None
    source_label: str | None = None
    doc_type: str | None = None
    status: str | None = None
    rule_applied_id: int | None = None
    dropbox_folder_id: int | None = None
    dropbox_file_path: str | None = None
    error_message: str | None = None
    processed_at: datetime


class ProcessResult(BaseModel):
    """Resumen de una pasada de procesado."""
    total: int = 0
    procesado: int = 0
    sin_regla: int = 0
    error: int = 0
    skipped: int = 0  # ya procesados en pasadas anteriores


class EmailStats(BaseModel):
    total: int = 0
    procesado: int = 0
    sin_regla: int = 0
    error: int = 0
